### ISSUE TITLE
internal/modules/cjs/loader.js:983   throw err;   ^  Could you help me with this code?

### ISSUE THREAD

**Issue by duol45** [nodejs/help#2644]
internal/modules/cjs/loader.js:983
  throw err;
  ^

Error: Cannot find module 'D:\Bryan Durán\Desktop\chat-javascript-fullstack-master\index.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)

    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74
:12)
    at internal/main/run_main_module.js:18:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []

**Comment by Romitou** [comment]
Hello, can you give more information?

**Comment by prasanthnrkrish** [comment]
I also get the same error on running
cd ios && pod install && cd ../ && react-native run-ios

Trying to get the react native up and running.

**Comment by lish90** [comment]
Hello,

 I'm having the same problem as stated above. When type {node index.js }  I get the error below, I tried number of solutions ie uninstalled node and npm then  reinstalled then back separately. I'm still getting the same error,  help needed. 
 
The error:
 ```
interactive-SVG-image git:(master) ✗ node index.js
internal/modules/cjs/loader.js:983
  throw err;
  ^

Error: Cannot find module '/Users/lish/code/JavaScript_projects/interactive-SVG-image/index.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

**Comment by shubhMjain** [comment]
delete node module and install again

**Comment by junhwong** [comment]
same errors:

```bash
...

> core-js-pure@3.6.5 postinstall /***/node_modules/core-js-pure
> node -e "try{require('./postinstall')}catch(e){}"


> ejs@2.7.4 postinstall /***/node_modules/ejs
> node ./postinstall.js

internal/modules/cjs/loader.js:969
  throw err;
  ^

Error: Cannot find module '/***/node_modules/ejs/postinstall.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
```


Tried many ways：

- rm -rf node_modules
- rm -f package-lock.jso
- npm cache clean --force
- reinstall nodejs
- ...

**Comment by junhwong** [comment]
versions:
  npm: '6.14.5',
  ares: '1.16.0',
  brotli: '1.0.7',
  cldr: '37.0',
  http_parser: '2.9.3',
  icu: '67.1',
  llhttp: '2.0.4',
  modules: '72',
  napi: '6',
  nghttp2: '1.41.0',
  node: '12.18.1',
  openssl: '1.1.1g',
  tz: '2019c',
  unicode: '13.0',
  uv: '1.38.0',
  v8: '7.8.279.23-node.38',
  zlib: '1.2.11'

**Comment by Tetr0** [comment]
Work for me: 
If you are trying to work with node (backend) & react native (fronted) try:

             1.  Create new main folder (mkdir rootProjectName)
             2. from ProjectName/backend > init the node side (npm init --yes )
             3. from ProjectName/> init the react side ( react-native init fronted )

:)

**Comment by shilpaanthati6** [comment]
> Hello,
> 
> I'm having the same problem as stated above. When type {node index.js } I get the error below, I tried number of solutions ie uninstalled node and npm then reinstalled then back separately. I'm still getting the same error, help needed.
> 
> The error:
> 
> ```
> interactive-SVG-image git:(master) ✗ node index.js
> internal/modules/cjs/loader.js:983
>  throw err;
>  ^
> 
> Error: Cannot find module '/Users/lish/code/JavaScript_projects/interactive-SVG-image/index.js'
>    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
>    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
>    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
>    at internal/main/run_main_module.js:18:47 {
>  code: 'MODULE_NOT_FOUND',
>  requireStack: []
> }
> ```

I do have the same issue .I tried many solutions ,I am feeling frustrated to solve this 🤣

**Comment by Tetr0** [comment]
Example index.js as module: 
***********************************************
 var productList = [{name: "banana"},{name: "apple"}
];

**module.exports =  { productList };**
***********************************************

**Comment by micstudent** [comment]
![hhhhhhhhhhhhhhhhhhhhhhhhhre](https://user-images.githubusercontent.com/57383005/88480019-5f97d580-cf79-11ea-9237-aeb12732349d.JPG)
_**_**Same Error! help**_**_

**Comment by micstudent** [comment]
> > Hello,
> > I'm having the same problem as stated above. When type {node index.js } I get the error below, I tried number of solutions ie uninstalled node and npm then reinstalled then back separately. I'm still getting the same error, help needed.
> > The error:
> > ```
> > interactive-SVG-image git:(master) ✗ node index.js
> > internal/modules/cjs/loader.js:983
> >  throw err;
> >  ^
> > 
> > Error: Cannot find module '/Users/lish/code/JavaScript_projects/interactive-SVG-image/index.js'
> >    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
> >    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
> >    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
> >    at internal/main/run_main_module.js:18:47 {
> >  code: 'MODULE_NOT_FOUND',
> >  requireStack: []
> > }
> > ```
> 
> I do have the same issue .I tried many solutions ,I am feeling frustrated to solve this 🤣

yes i Fix it

1. `npm cache clean --force`

2. ### delete **node_modules** _folder_

3. ### delete **package-lock.json** _file_

4. `npm install`

**Comment by adogam** [comment]
I am struggling with this same issue. I followed the fix as well and it is not working for me. I am on windows 10 if that makes any difference. It seems to not be downloading the complete fileset of Node. Any help would be amazing.

**Comment by vinodkv007** [comment]
same issue, tried all of the solutions mentioned above, but no use.
``
C:\Users\vinod\server>npm start

> server@1.0.0 start C:\Users\vinod\server
> nodemon expGqlServer.js

[nodemon] 2.0.4
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node expGqlServer.js`
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module '../../temp'
Require stack:
- C:\Users\vinod\server\resolvers\post.js
- C:\Users\vinod\server\node_modules\@graphql-toolkit\file-loading\index.cjs.js
- C:\Users\vinod\server\node_modules\merge-graphql-schemas\index.cjs.js
- C:\Users\vinod\server\expGqlServer.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (C:\Users\vinod\server\resolvers\post.js:2:14)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\vinod\\server\\resolvers\\post.js',
    'C:\\Users\\vinod\\server\\node_modules\\@graphql-toolkit\\file-loading\\index.cjs.js',
    'C:\\Users\\vinod\\server\\node_modules\\merge-graphql-schemas\\index.cjs.js',
    'C:\\Users\\vinod\\server\\expGqlServer.js'
  ]
}
[nodemon] app crashed - waiting for file changes before starting... ``

**Comment by adogam** [comment]
> same issue, tried all of the solutions mentioned above, but no use.
> ``
> C:\Users\vinod\server>npm start
> 
> > server@1.0.0 start C:\Users\vinod\server
> > nodemon expGqlServer.js
> 
> [nodemon] 2.0.4
> [nodemon] to restart at any time, enter `rs`
> [nodemon] watching path(s): _._
> [nodemon] watching extensions: js,mjs,json
> [nodemon] starting `node expGqlServer.js`
> internal/modules/cjs/loader.js:968
> throw err;
> ^
> 
> Error: Cannot find module '../../temp'
> Require stack:
> 
> * C:\Users\vinod\server\resolvers\post.js
> * C:\Users\vinod\server\node_modules@graphql-toolkit\file-loading\index.cjs.js
> * C:\Users\vinod\server\node_modules\merge-graphql-schemas\index.cjs.js
> * C:\Users\vinod\server\expGqlServer.js
>   at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
>   at Function.Module._load (internal/modules/cjs/loader.js:841:27)
>   at Module.require (internal/modules/cjs/loader.js:1025:19)
>   at require (internal/modules/cjs/helpers.js:72:18)
>   at Object. (C:\Users\vinod\server\resolvers\post.js:2:14)
>   at Module._compile (internal/modules/cjs/loader.js:1137:30)
>   at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
>   at Module.load (internal/modules/cjs/loader.js:985:32)
>   at Function.Module._load (internal/modules/cjs/loader.js:878:14)
>   at Module.require (internal/modules/cjs/loader.js:1025:19) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [
>   'C:\Users\vinod\server\resolvers\post.js',
>   'C:\Users\vinod\server\node_modules\@graphql-toolkit\file-loading\index.cjs.js',
>   'C:\Users\vinod\server\node_modules\merge-graphql-schemas\index.cjs.js',
>   'C:\Users\vinod\server\expGqlServer.js'
>   ]
>   }
>   [nodemon] app crashed - waiting for file changes before starting... ``

I fixed the issue by using and trying different versions of node. It took a while to find the right one but I eventually did.

**Comment by vinodkv007** [comment]
I fixed the issue, apparently whenever you change your project directory some node modules will be imported automatically by node.js and along with it, some file paths inside the project will be changed.
In my case when I moved my project to a different folder, 
``
 ../temp
``
was changed to,
``
../../temp
``
I deleted the `` ../`` and now it's working fine.

**Comment by mrrobotja** [comment]
Where did you find the temp file?

**Comment by yonas1104** [comment]
node js can only run app.js like you execute main function in other programming languages!

**Comment by sefrem** [comment]
To anyone who stumbled upon this issue. 
All the above haven't worked for me, while deleting all the whitespaces from the path worked perfectly. 
As i see, topicstarter also has a whitespace in the path. This might've been the issue.

**Comment by maherbenissa** [comment]
you will just run this command line :

rm -rf node_modules package-lock.json && npm install && npm start

**Comment by allocious** [comment]
Am also facing the same issue... here is the solution from my end.
if index.js not in our root folder means, we will getting error, once you move to root folder then server will be running

**Comment by naimur978** [comment]
> Am also facing the same issue... here is the solution from my end.
> if index.js not in our root folder means, we will getting error, once you move to root folder then server will be running

Worked for me.

**Comment by shilpaanthati6** [comment]
I too got faced the same issue . I got rid that finally by removing node
modules  that got installed twice and twice.

**Comment by barhouum7** [comment]
I got faced the same issue and I solved it finally.
In my case I was performing `git push heroku main` which also `main` isn't the default branch.
I had to go to master branch first(default branch), merge the code with my other branch and then `heroku git:remote -a <repoName>` and `git push heroku master`.

HOPE this can help!

**Comment by abhiraizada** [comment]
Oh my god!! i was struggling on this, tried everything, in the end, found out my file name was sever.js

**Comment by PoojaDurgad** [comment]
@duol45 - is there anything pending in this issue?

**Comment by deepakyadav2254** [comment]
If someone still stuck, leave a comment.

**Comment by MOSMOSterTVZall** [comment]
2020-10-13T04:57:12.561758+00:00 app[Worker.1]: internal/modules/cjs/loader.js:883
2020-10-13T04:57:12.561775+00:00 app[Worker.1]:   throw err;
2020-10-13T04:57:12.561775+00:00 app[Worker.1]:   ^
2020-10-13T04:57:12.561776+00:00 app[Worker.1]: 
2020-10-13T04:57:12.561776+00:00 app[Worker.1]: Error: Cannot find module './config.json'
2020-10-13T04:57:12.561776+00:00 app[Worker.1]: Require stack:
2020-10-13T04:57:12.561777+00:00 app[Worker.1]: - /app/index.js
2020-10-13T04:57:12.561778+00:00 app[Worker.1]:     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
2020-10-13T04:57:12.561778+00:00 app[Worker.1]:     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
2020-10-13T04:57:12.561778+00:00 app[Worker.1]:     at Module.require (internal/modules/cjs/loader.js:952:19)
2020-10-13T04:57:12.561779+00:00 app[Worker.1]:     at require (internal/modules/cjs/helpers.js:88:18)
2020-10-13T04:57:12.561779+00:00 app[Worker.1]:     at Object.<anonymous> (/app/index.js:7:20)
2020-10-13T04:57:12.561780+00:00 app[Worker.1]:     at Module._compile (internal/modules/cjs/loader.js:1063:30)
2020-10-13T04:57:12.561780+00:00 app[Worker.1]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
2020-10-13T04:57:12.561780+00:00 app[Worker.1]:     at Module.load (internal/modules/cjs/loader.js:928:32)
2020-10-13T04:57:12.561781+00:00 app[Worker.1]:     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
2020-10-13T04:57:12.561781+00:00 app[Worker.1]:     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12) {
2020-10-13T04:57:12.561782+00:00 app[Worker.1]:   code: 'MODULE_NOT_FOUND',
2020-10-13T04:57:12.561782+00:00 app[Worker.1]:   requireStack: [ '/app/index.js' ]
2020-10-13T04:57:12.561783+00:00 app[Worker.1]: }

**Comment by PoojaDurgad** [comment]
@MOSMOSterTVZall - can you elaborate your issue more please? which command you are trying to run ? 
for me by looking at the error log your issue looks like there is a missing config.json in your current working path. make sure it is configured properly and if that is not the case try to add explicitly to the package.json hope it may help.

**Comment by varunreddy305** [comment]
> > > Hello,
> > > I'm having the same problem as stated above. When type {node index.js } I get the error below, I tried number of solutions ie uninstalled node and npm then reinstalled then back separately. I'm still getting the same error, help needed.
> > > The error:
> > > ```
> > > interactive-SVG-image git:(master) ✗ node index.js
> > > internal/modules/cjs/loader.js:983
> > >  throw err;
> > >  ^
> > > 
> > > Error: Cannot find module '/Users/lish/code/JavaScript_projects/interactive-SVG-image/index.js'
> > >    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
> > >    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
> > >    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
> > >    at internal/main/run_main_module.js:18:47 {
> > >  code: 'MODULE_NOT_FOUND',
> > >  requireStack: []
> > > }
> > > ```
> > 
> > 
> > I do have the same issue .I tried many solutions ,I am feeling frustrated to solve this 🤣
> 
> yes i Fix it
> 
> 1. `npm cache clean --force`
> 2. ### delete **node_modules** _folder_
> 3. ### delete **package-lock.json** _file_
> 4. `npm install`

Worked for me thanks

**Comment by dollychaudhary54** [comment]
Please HELP!!! I have tried deleting the node_modules and json files multiple times. Nothing working for me.

$ nodemon script.js
[nodemon] 2.0.6
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node script.js script.js`
internal/modules/cjs/loader.js:834
  throw err;
  ^

Error: Cannot find module 'C:\Users\Ankur\Desktop\node-website\script.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:831:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
[nodemon] app crashed - waiting for file changes before starting...

**Comment by Erisjimver** [comment]
Interesante, me salia el mismo error y era porque no estaba ejecutando mi js desde la carpeta del archivo.

**Comment by Samira-hm-a** [comment]
Worked for with this solution:


> > > > Hello,
> > > > I'm having the same problem as stated above. When type {node index.js } I get the error below, I tried number of solutions ie uninstalled node and npm then reinstalled then back separately. I'm still getting the same error, help needed.
> > > > The error:
> > > > ```
> > > > interactive-SVG-image git:(master) ✗ node index.js
> > > > internal/modules/cjs/loader.js:983
> > > >  throw err;
> > > >  ^
> > > > 
> > > > Error: Cannot find module '/Users/lish/code/JavaScript_projects/interactive-SVG-image/index.js'
> > > >    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
> > > >    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
> > > >    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
> > > >    at internal/main/run_main_module.js:18:47 {
> > > >  code: 'MODULE_NOT_FOUND',
> > > >  requireStack: []
> > > > }
> > > > ```
> > > 
> > > 
> > > I do have the same issue .I tried many solutions ,I am feeling frustrated to solve this 🤣
> > 
> > 
> > yes i Fix it
> > 
> > 1. `npm cache clean --force`
> > 2. ### delete **node_modules** _folder_
> > 3. ### delete **package-lock.json** _file_
> > 4. `npm install`
> 
> Worked with this solution! Thank you

**Comment by Niloy1010** [comment]
This is probably really dumb but for me, the problem was vs code automatically added an import to the top without me noticing. And that was not a part of my project nor it was something I was using. Simply removing it fixed my issue. There is less than 5% chance that this is the issue but someone watching please do check your imports.

**Comment by mmartinsolisdev** [comment]
I was trying for an hour doing everything they say but it didn't work, it worked when I created an index.js file without deleting my app.js file, then I went into the package.json file and made a change and saved, then run nodemon again and everything was fine. Then I left the file as it was originally and had no problem, it was weird.

**Comment by Abdur-Razzak-Rouzex** [comment]
Whenever you are creating your react app, follow the following steps, hope you will not face this kind of problems any further:

Create a folder to hold the React Application and open the command prompt as administrator inside the folder:

`npx create-react-app app_name ./ ` (use dot and the /)
`cd app_name` 
`npm start`

**Comment by shyama-ui** [comment]
> Am also facing the same issue... here is the solution from my end.
> if index.js not in our root folder means, we will getting error, once you move to root folder then server will be running

Thanks a ton!! Was trying everything but this worked !!!!  👍🏻 Thanks a ton!

**Comment by Samk13** [comment]
Same problem: 
run webpack-dev-server
```zsh

> webpack-ts@1.0.0 serve D:\Workspace\code\webpack-ts
> webpack-dev-server

internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'webpack-cli/bin/config-yargs'
Require stack:
- D:\Workspace\code\webpack-ts\node_modules\webpack-dev-server\bin\webpack-dev-server.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (D:\Workspace\code\webpack-ts\node_modules\webpack-dev-server\bin\webpack-dev-server.js:65:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\Workspace\\code\\webpack-ts\\node_modules\\webpack-dev-server\\bin\\webpack-dev-server.js'
  ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! webpack-ts@1.0.0 serve: `webpack-dev-server`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the webpack-ts@1.0.0 serve script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\myname\AppData\Roaming\npm-cache\_logs\2020-12-15T23_17_25_326Z-debug.log

```

I tried `npm cache clean --force` after deleting node_modules and package-lock with no luck.🤷🏽‍♂️

**Comment by sesaqib** [comment]
This "https://stackoverflow.com/questions/14949118/node-js-error-cannot-find-module-express" link can help you.

**Comment by N0mercy00** [comment]
same error : 
npm run start then. 883 error make me sad :(

>PS D:\D_desktop\study\REACT_BOILERPLATE> npm run start
>\> react_boilerplate@1.0.0 start D:\D_desktop\study\REACT_BOILERPLATE
> \>node ./server/index.js
>internal/modules/cjs/loader.js:883
 > throw err;
  ^
>Error: Cannot find module './server/models/User'
>Require stack:

i tried changed path , and tried delete cash, node_modules , install npm again but still not fix :(
Has anyone solved 883 problem?

**Comment by ahmedmaria** [comment]
> This is probably really dumb but for me, the problem was vs code automatically added an import to the top without me noticing. And that was not a part of my project nor it was something I was using. Simply removing it fixed my issue. There is less than 5% chance that this is the issue but someone watching please do check your imports.

For me, I had "const express = require('express');" somehow vscode added "const express = require('./node_modules/express');". I had to delete ./node_modules part and it resolved the issue. I got the idea of checking that from you. Thanks :)

**Comment by bdamor5** [comment]
internal/modules/cjs/loader.js:968
  throw err;
  ^

internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module 'D:\SocialApp(backend)\index.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
Facing the same error.Also , index.js is not present in the root folder and working on backend.Please help me solve this issue.Thank You.

**Comment by CarlosLasprilla** [comment]
Tenía el mismo problema: después de instalar npm install react-router-dom - Luego iniciaba el servidor con npm start obtenía el siguiete error:

Error: Cannot find module 'react'
Require stack:
- C:\Users\Usuario\AppData\Roaming\npm\node_modules\react-scripts\scripts\start.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.resolve (internal/modules/cjs/helpers.js:94:19)
    at Object.<anonymous> (C:\Users\Usuario\AppData\Roaming\npm\node_modules\react-scripts\scripts\start.js:52:31)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
  ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! front-end@0.1.0 start: `react-scripts start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the front-end@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

Lo solucioné instalando todo la librería npm, es decir instalando npm install.

Espero esto te sirva

**Comment by SwastikGowdaL** [comment]
hey guys i am also getting the same problem when i try to use webpack - 
it would be really helpful if anyone could help out here - 

![image](https://user-images.githubusercontent.com/68393994/103848448-65691980-50c8-11eb-8989-18277cbaad7d.png)

**Comment by SwastikGowdaL** [comment]
> @MOSMOSterTVZall - can you elaborate your issue more please? which command you are trying to run ?
> for me by looking at the error log your issue looks like there is a missing config.json in your current working path. make sure it is configured properly and if that is not the case try to add explicitly to the package.json hope it may help.

![image](https://user-images.githubusercontent.com/68393994/103848577-b2e58680-50c8-11eb-84bc-9e092fe908b5.png)

can you help to solve this ?

**Comment by Lxxyx** [comment]
> > @MOSMOSterTVZall - can you elaborate your issue more please? which command you are trying to run ?
> > for me by looking at the error log your issue looks like there is a missing config.json in your current working path. make sure it is configured properly and if that is not the case try to add explicitly to the package.json hope it may help.
> 
> ![image](https://user-images.githubusercontent.com/68393994/103848577-b2e58680-50c8-11eb-84bc-9e092fe908b5.png)
> 
> can you help to solve this ?

`npm install v8-compile-cache` or delete node modules and reinstall.

If that doesn't work, you need to search for code that uses `v8-compile-cache` and commit the issue to that npm module

**Comment by BarrySougue** [comment]
moi j'ai taper une commande npm run dev sur la terminale de VSCodes mais on m'affiche ces erreurs.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! frontend@1.0.0 dev: `webpack --mode development --watch`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the frontend@1.0.0 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Barry&Sougué\AppData\Roaming\npm-cache\_logs\2021-01-07T15_16_05_462Z-debug.log
PS C:\Users\Barry&Sougué\Desktop\Gestion-De-Restaurant\Restaurant\frontend>
j'ai essayer beaucoup des choses jusqu'à présent j'ai pas de solution, stp j'ai besoin d'aide.

**Comment by saadBinZahoor** [comment]
> **First, check the path of your current folder under terminal**
> else you will just run this command line :
> 
> rm -rf node_modules package-lock.json && npm install && npm start

This method worked for me

**Comment by jianleee** [comment]
I have the same error:

Error: Cannot find module 'D:\Learn New Tech\Learn Nodejs\index.js'
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

After i rewrite the path in srcipt package.jon to right path it works, so you may try rewrite the path, hope this can help

**Comment by MobackChetanKumar** [comment]
internal/modules/cjs/loader.js:834
  throw err;
  ^

Error: Cannot find module 'C:\Projects\client-stage-fe\dist\server\main.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:831:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bookcab@0.0.0 serve:ssr: `node dist/server/main.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the client@0.0.0 serve:ssr script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\pc\AppData\Roaming\npm-cache\_logs\2021-02-09T08_34_40_367Z-debug.log

Tried several times deleting node modules and install again with admin permission. Still not working. any help appreciated

**Comment by Jubintgh** [comment]
I recently ran into this issue and I realized the solution was pretty simple, 
1. make sure you're in the right Directory from terminal and, 
2. also be sure to save your file and its changes before running node

**Comment by Teetomania** [comment]
If you're still facing the issue, use "nvm install latest" the error should disappear after updating to the latest node and npm versions and switching to it.

**Comment by iluyobrainy** [comment]
Hi N0mercy00, all you have to do to solve the 883 problem 
1. You should cut (ctrl + x ) your main file (build.js, index.js , server.js) from the folder you saved it in it is in and paste it in your main folder that houses all the subfolders then after that you can re-run node 

 OR

 IF THAT DOESNT WORK OR YOU DONT WANT TO GO THROUGH ALL THAT STRESS 

2. Just cd (change directory) to you main file((build.js, index.js , server.js) if it in a folder then re-run node

> same error :
> npm run start then. 883 error make me sad :(
> 
> > PS D:\D_desktop\study\REACT_BOILERPLATE> npm run start
> > > react_boilerplate@1.0.0 start D:\D_desktop\study\REACT_BOILERPLATE
> > >node ./server/index.js
> > internal/modules/cjs/loader.js:883
> > throw err;
> > ^
> > Error: Cannot find module './server/models/User'
> > Require stack:
> 
> i tried change path , and tried delete cache, node_modules , install npm again but still not fix :(
> Has anyone solved 883 problem?

**Comment by Vi1234sh12** [comment]
Anyone, please can you help me for solving this problem 
![Screenshot (575)](https://user-images.githubusercontent.com/45570993/110112797-33c6b300-7dd8-11eb-8cdc-d967d7ee4af6.png)

**Comment by karuifeather** [comment]
Similar issue.

Was trying to start a new Next.js project. Installation failed with the following error. Right after I got that error, both yarn and npm stopped working and show the same error. Even when I run something like `npm --version`, it continues to fail. 

I get the same error irrespective of where I'm running commands from. An hour has passed since and I continue to be stuck. Everything was working just fine an hour ago. I'm on Windows 10. Node itself works and its version is 14.15.5.

I tried reinstalling but unfortunately for me, that didn't fix the issue.

````$ npm --version
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'C:\Users\USER_NAME\AppData\Roaming\npm\node_modules\npm\bin\npm-cli.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}````

**Comment by jadavneil** [comment]
I also had an issue and it was throwing me the same error, so I went to my other project folder and tried over there by ```npm start``` and surprisingly it worked there so I looked in the folder structure and I found out that in my folder I was using the "@" special character, believe me, I changed the folder name and again hit the command ```npm start``` and it worked for me.

**Comment by marsonya** [comment]
> I also had an issue and it was throwing me the same error, so I went to my other project folder and tried over there by `npm start` and surprisingly it worked there so I looked in the folder structure and I found out that in my folder I was using the "@" special character, believe me, I changed the folder name and again hit the command `npm start` and it worked for me.
> 
> @duol45 believe me change your folder name as there is one special character is present `á` change it to normal `a`. And it should work.

@nodejs/modules, it seems the that the presence of special characters in path fails to load modules.
I am not sure if this is an issue or not. Just thought you might want to take a look at this.

**Comment by Arunaa2783** [comment]
> If someone still stuck, leave a comment.

yes I am .. my error not solved yet

**Comment by DerekNonGeneric** [comment]
> @nodejs/modules, [&hellip;] presence of special characters in path fails to load modules.

Tested whether module paths containing `á` would resolve using v14.16.0 on Debian and could not reproduce this error locally. Perhaps it has already been fixed, but please [file an issue in the core repo](https://github.com/nodejs/node/issues/new?assignees=&labels=&template=1-bug-report.md) if anyone is able to repro (the info requested by the issue template would help a lot).

**Comment by kingmalik100** [comment]
[nodemon] 2.0.6
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node index.js`
internal/modules/cjs/loader.js:1083
  throw err;
  ^

Error: Cannot find module '/Users/kingmalik/Documents/Node/RoutesPro/index.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1080:15)
    at Function.Module._load (internal/modules/cjs/loader.js:923:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
[nodemon] app crashed - waiting for file changes before starting...
[nodemon] restarting due to changes...
[nodemon] starting `node index.js`
internal/modules/cjs/loader.js:1083
  throw err;
  ^

Error: Cannot find module '/Users/kingmalik/Documents/Node/RoutesPro/index.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1080:15)
    at Function.Module._load (internal/modules/cjs/loader.js:923:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
[nodemon] app crashed - waiting for file changes before starting...

**Comment by Imprasna-signatures1** [comment]
D:\Tailwind>npm run build:css

> Tailwind@1.0.0 build:css D:\Tailwind
> tailwind build src/style.css -o dist/styles.css

internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'autoprefixer'
Require stack:
- D:\Tailwind\node_modules\tailwindcss\lib\cli\commands\build.js
- D:\Tailwind\node_modules\tailwindcss\lib\cli\commands\index.js
- D:\Tailwind\node_modules\tailwindcss\lib\cli\main.js
- D:\Tailwind\node_modules\tailwindcss\lib\cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (D:\Tailwind\node_modules\tailwindcss\lib\cli\commands\build.js:9:44)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\Tailwind\\node_modules\\tailwindcss\\lib\\cli\\commands\\build.js',
    'D:\\Tailwind\\node_modules\\tailwindcss\\lib\\cli\\commands\\index.js',
    'D:\\Tailwind\\node_modules\\tailwindcss\\lib\\cli\\main.js',
    'D:\\Tailwind\\node_modules\\tailwindcss\\lib\\cli.js'
  ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! Tailwind@1.0.0 build:css: `tailwind build src/style.css -o dist/styles.css`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the Tailwind@1.0.0 build:css script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Dell\AppData\Roaming\npm-cache\_logs\2021-04-21T09_43_35_052Z-debug.log

**Comment by Imprasna-signatures1** [comment]
Can someone help  me figure out what's the error and how do I fix it?

**Comment by Daniel9756** [comment]
I have the same issue too. I was working on ts, typeorm, node.js, and express.
But I did the following and cleared the error.
 
npm cache clean --force
delete  the node module folder
delete the package-lock.json
npm i

**Comment by jsbimra** [comment]
> I have the same issue too. I was working on ts, typeorm, node.js, and express.
> But I did the following and cleared the error.
> 
> npm cache clean --force
> delete the node module folder
> delete the package-lock.json
> npm i

Thanks this worked!

**Comment by taqwaaziz95** [comment]
**> delete node module and install again

anjir solusi bangget lu !**

**Comment by haileemiu** [comment]
For me it was an xcode issue. 

Remove xcode cli tools: `sudo rm -r -f /Library/Developer/CommandLineTools`
Reinstall: `xcode-select --install`
Then, delete node modules. Manually or with `rm -rf node_modules`
Should be good to go: `npm install` then `npm start`

https://github.com/haileemiu/Notes/blob/main/Troubleshooting/npm.md

**Comment by Jood80** [comment]
I'm using Nest.js and this was my error
```ts
node:internal/modules/cjs/loader:930
  throw err;
  ^

Error: Cannot find module '/home/inspiron/Documents/ZTM/nestJs/blog-backend/dist/main'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:927:15)
    at Function.Module._load (node:internal/modules/cjs/loader:772:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:79:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```
I tried 
> rm -rf node_modules package-lock.json dist
> npm install
> npm run start:dev

which didn't work out. also re-installing node didn't do the trick either.
**In my case, the problem was** that my `main.ts` file was not in the root of `src` folder, instead it was in `src/modules/main.ts`.
the problem's been solved when I moved it back to `src` directory
![image](https://user-images.githubusercontent.com/56412800/128854184-337d15cb-3bb5-42a6-a2a1-590aaefcf2e8.png)

**Comment by FareenaImran786** [comment]
![image](https://user-images.githubusercontent.com/92385415/146020790-57c92dfb-658d-48e6-ba28-7828826abe69.png)

Hi, I just tried to downgrade the appium version with these command lines
npm uninstall -g appium
npm install -g appium@1.6.1

and then got this error. Can anyone help me to resolve this please?

**Comment by GantMan** [comment]
Like https://github.com/nodejs/help/issues/2644#issuecomment-628579784 said

I went to my global node_modules and just wiped them out.  Now it works.

**Comment by nimmyranjee** [comment]
Create the .js file in the root folder .

**Comment by Dezzign** [comment]
Here's what ended up working for me (Strapi v3.6.8)

None of the above worked ...
(npm cache clean --farce, deleting node_modules, package-lock.json etc)

Neither did moving server.js files around etc.

Bizarrely, just using **this start command worked**: strapi start

Somehow that's different to running: node server.js etc

Here's my package.json start scripts for reference:

```
"scripts": {
      "develop": "strapi develop",
      "development": "cross-env NODE_ENV=development npm run strapi develop",
      "start": "strapi start",
      "build": "cross-env NODE_ENV=production npm run build",
      "gcp-build": "strapi build",
      "strapi": "strapi start"
  },
```

**Comment by charu510** [comment]
Simply use the cd command for going to the specific directory where the script is written. It will have to be done manually as the command is not being able to direct properly. Once done the script runs successfully. This method worked for me.

**Comment by Ravi310893** [comment]
> internal/modules/cjs/loader.js:983 throw err; ^
> 
> Error: Cannot find module 'D:\Bryan Durán\Desktop\chat-javascript-fullstack-master\index.js' at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
> 
> ```
> at Function.Module._load (internal/modules/cjs/loader.js:862:27)
> at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74
> ```
> 
> :12) at internal/main/run_main_module.js:18:47 { code: 'MODULE_NOT_FOUND', requireStack: []

check the address when giving command like if you giving    node  .\module  ,  check if its inside src and provide src before module name like  node .\src\module

**Comment by ShahnwazA78** [comment]
first check whether in the json file the dependencies you are require for is mentioned there or not after your installation, in my case it was not listed there so i double check and it was not listed as just becouse i did one silly mistake that the location to install dependencies is different where the src folder so that is why the dependencies were not listed there i just reinstall and its working fine.

**Comment by yandyestrada** [comment]
Hello I having issues with this please help


node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module '/Users/yandyestrada/ethers_examples/start'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.14.2
yandyestrada@Yandys-MBP-2 ethers_examples %

**Comment by RamiE965** [comment]
> Hello I having issues with this please help
> 
> node:internal/modules/cjs/loader:1078 throw err; ^
> 
> Error: Cannot find module '/Users/yandyestrada/ethers_examples/start' at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15) at Module._load (node:internal/modules/cjs/loader:920:27) at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) at node:internal/main/run_main_module:23:47 { code: 'MODULE_NOT_FOUND', requireStack: [] }
> 
> Node.js v18.14.2 yandyestrada@Yandys-MBP-2 ethers_examples %

I have the same exact issue, did you find a fix?

**Comment by KRISHNO-001** [comment]
Check if Nodejs directory(generally -> C:\Program Files\nodejs) location is added in the system variables or not  if not then add  and restart everything 
Hope this fixes the issue

**Comment by Farhaan-Hash** [comment]
> Am also facing the same issue... here is the solution from my end. if index.js not in our root folder means, we will getting error, once you move to root folder then server will be running

This was simple thanks. Worked

**Comment by aastha01chauhan** [comment]
for Linux users: rm -rf node_modules package-lock.json && npm install && npm start
This worked for me

**Comment by Deep7452** [comment]
Step 1. press Window +r
Step 2. And type %AppData%  and press Enter
Step 3. you find there npn folder ...Delete this folder and run the command  "npm install -g npm@latest"

And if you can do this your project again run smoothly.
Thank you

**Comment by domajor** [comment]
Ubuntu Server 20.04 LTS

/init.js:27

SyntaxError: Unexpected token '.'
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47

Node.js v18.17.1

:( :( :(

**Comment by ljharb** [comment]
@domajor yours is probably just JS syntax that node 18 doesn't understand.

**Comment by domajor** [comment]
Ubuntu reinstalled, node.js reinstalled.
rm -rf node_modules package-lock.json && npm install && npm start ; was established , 

node init.js
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'dateformat'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/root/pool/lib/logger.js:11:18)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)

**Comment by domajor** [comment]
I tried 1 or 2 months ago and the working system died after updating, nothing is working

**Comment by domajor** [comment]
It worked when the server was restarted and updated, I wonder how long it will work.

**Comment by domajor** [comment]
nodejs :) ubuntu server 20.04 
[master] Starting Cryptonote Node.JS pool version v2.0.0
[master] Invalid pool wallet address in configuration file (poolServer.poolAddress)

[master] Starting Cryptonote Node.JS pool version Oguzhan
[master] yagdi yagmur cakti simsek , bendemi sair olsam acep  (OzanServer.FevziAddress)

unresolved error
Does anyone have any advice?
tavsiyesi olan var mı ?

**Comment by domajor** [comment]
[master] Starting Cryptonote Node.JS pool version v2.0.0
(node:2240) UnhandledPromiseRejectionWarning: Error: The client is closed

I'm so tired, I have no hope :( I'm using Ubuntu Server 20

**Comment by leyroy** [comment]
> [master] Starting Cryptonote Node.JS pool version v2.0.0 (node:2240) UnhandledPromiseRejectionWarning: Error: The client is closed
> 
> I'm so tired, I have no hope :( I'm using Ubuntu Server 20

this kind of is breaking me 😢

**Comment by kabornblack** [comment]
I am facing some issues when trying to run sass: I have tried all the above mentioned solutions and nothing seems to work so far.

PS C:\Users\admin\Desktop\IT\JONAS CSS & SASS\Natours-starter> npm run compile:sass

> natours-starter@1.0.0 compile:sass
> node-sass sass/main.scss starter/css/style.css

'SASS\Natours-starter\node_modules\.bin\' is not recognized as an internal or external command,
operable program or batch file.
node:internal/modules/cjs/loader:998
  throw err;
  ^

Error: Cannot find module 'C:\Users\admin\Desktop\IT\node-sass\bin\node-sass'
    at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
    at Module._load (node:internal/modules/cjs/loader:841:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.12.1
PS C:\Users\admin\Desktop\IT\JONAS CSS & SASS\Natours-starter>

**Comment by domajor** [comment]
ubuntu server 
 node init.js
(node:40292) UnhandledPromiseRejectionWarning: Error: The client is closed
 Commander._RedisClient_sendCommand
errors continue

**Comment by YoussefElogail** [comment]
I faced this problem today and it was because of the path name. It was called redux and it was working. Then I changed the file to redux & react-redux and the problem appeared. The problem was in & for me.

**Comment by YoussefElogail** [comment]
> I am facing some issues when trying to run sass: I have tried all the above mentioned solutions and nothing seems to work so far.
> 
> PS C:\Users\admin\Desktop\IT\JONAS CSS & SASS\Natours-starter> npm run compile:sass
> 
> > natours-starter@1.0.0 compile:sass
> > node-sass sass/main.scss starter/css/style.css
> 
> 'SASS\Natours-starter\node_modules.bin' is not recognized as an internal or external command, operable program or batch file. node:internal/modules/cjs/loader:998 throw err; ^
> 
> Error: Cannot find module 'C:\Users\admin\Desktop\IT\node-sass\bin\node-sass' at Module._resolveFilename (node:internal/modules/cjs/loader:995:15) at Module._load (node:internal/modules/cjs/loader:841:27) at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) at node:internal/main/run_main_module:23:47 { code: 'MODULE_NOT_FOUND', requireStack: [] }
> 
> Node.js v18.12.1 PS C:\Users\admin\Desktop\IT\JONAS CSS & SASS\Natours-starter>

remove & from JONAS CSS & SASS => JONAS CSS and SASS

**Comment by MohamedNashad** [comment]
npm run dev

> workment@2.0.0 dev     
> tsc && nodemon ./app.ts

[nodemon] 2.0.22
[nodemon] to restart at any time, enter `rs`      
[nodemon] watching path(s): src\**\*
[nodemon] watching extensions: ts,js
[nodemon] starting `ts-node ./src/app.ts ./app.ts`
node:internal/modules/cjs/loader:1147
  throw err;
  ^

Error: Cannot find module './app.ts'
Require stack:
- E:\Projects\iconblocks\workedn\wp-workedn-server\src\imaginaryUncacheableRequireResolveScript
    at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at Function.resolve (node:internal/modules/helpers:187:19)
    at requireResolveNonCached (E:\Projects\iconblocks\workedn\wp-workedn-server\node_modules\ts-node\dist\bin.js:549:16)      
    at getProjectSearchDir (E:\Projects\iconblocks\workedn\wp-workedn-server\node_modules\ts-node\dist\bin.js:519:40)
    at phase3 (E:\Projects\iconblocks\workedn\wp-workedn-server\node_modules\ts-node\dist\bin.js:267:27)
    at bootstrap (E:\Projects\iconblocks\workedn\wp-workedn-server\node_modules\ts-node\dist\bin.js:47:30)
    at main (E:\Projects\iconblocks\workedn\wp-workedn-server\node_modules\ts-node\dist\bin.js:33:12)
    at Object.<anonymous> (E:\Projects\iconblocks\workedn\wp-workedn-server\node_modules\ts-node\dist\bin.js:579:5)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'E:\\Projects\\iconblocks\\workedn\\wp-workedn-server\\src\\imaginaryUncacheableRequireResolveScript'
  ]
}

Node.js v20.11.1
[nodemon] app crashed - waiting for file changes before starting...


i have the same issue, even though i most of the solutions above, the app.ts file is available in my root directory, still i am getting the error

**Comment by aloysiusose** [comment]
> ![image](https://private-user-images.githubusercontent.com/127918448/257068544-2cb6da19-bd38-4a62-a269-b9338eadced2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5NjI4OTgsIm5iZiI6MTcxMDk2MjU5OCwicGF0aCI6Ii8xMjc5MTg0NDgvMjU3MDY4NTQ0LTJjYjZkYTE5LWJkMzgtNGE2Mi1hMjY5LWI5MzM4ZWFkY2VkMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMFQxOTIzMThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MDY5ZmUxMTM4ZDJmYTA4YTYyNzMyZjgzMzZiYWIzMTQyMWJiYzdmMzgyNjE5OWEzMmVhNzBmOWRhMzM3M2ExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.lx6RsFhbHopIRus2o1DnF0Pw45-brOzwpUIjLSOJx2k)
> 
> **This is helpful for this issue**
> 
> Step 1. press Window +r Step 2. And type %AppData% and press Enter Step 3. you find there npn folder ...Delete this folder and run the command "npm install -g npm@latest"
> 
> And if you can do this your project again run smoothly. Thank you

Thanks for this!!

**Comment by sureshdhwaniris123** [comment]
Node version : 14.17.5
Azure azure/storage-blob : ^12.17.0

internal/modules/cjs/loader.js:892

throw err;

^

Error: Cannot find module 'node:os'

Require stack:

- /home/suresh-dhwani/citifinance-node-api/node_modules/@azure/logger/dist/commonjs/log.js

- /home/suresh-dhwani/citifinance-node-api/node_modules/@azure/logger/dist/commonjs/debug.js

- /home/suresh-dhwani/citifinance-node-api/node_modules/@azure/logger/dist/commonjs/index.js

- /home/suresh-dhwani/citifinance-node-api/node_modules/@azure/core-http/dist/index.js

- /home/suresh-dhwani/citifinance-node-api/node_modules/@azure/storage-blob/dist/index.js

- /home/suresh-dhwani/citifinance-node-api/service/blob_service.js

- /home/suresh-dhwani/citifinance-node-api/service/getBlobUrl.js

- /home/suresh-dhwani/citifinance-node-api/service/common.js

- /home/suresh-dhwani/citifinance-node-api/service/index.js

- /home/suresh-dhwani/citifinance-node-api/app.js

- /home/suresh-dhwani/citifinance-node-api/server.js

at Function.Module._resolveFilename (internal/modules/cjs/loader.js:889:15)

at Function.Module._load (internal/modules/cjs/loader.js:745:27)

at Module.require (internal/modules/cjs/loader.js:961:19)

at require (internal/modules/cjs/helpers.js:92:18)

at Object.<anonymous> (/home/suresh-dhwani/citifinance-node-api/node_modules/@azure/logger/dist/commonjs/log.js:7:19)

at Module._compile (internal/modules/cjs/loader.js:1072:14)

at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)

at Module.load (internal/modules/cjs/loader.js:937:32)

at Function.Module._load (internal/modules/cjs/loader.js:778:12)

at Module.require (internal/modules/cjs/loader.js:961:19)

at require (internal/modules/cjs/helpers.js:92:18)

at Object.<anonymous> (/home/suresh-dhwani/citifinance-node-api/node_modules/@azure/logger/dist/commonjs/debug.js:5:18)

at Module._compile (internal/modules/cjs/loader.js:1072:14)

at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)

at Module.load (internal/modules/cjs/loader.js:937:32)

at Function.Module._load (internal/modules/cjs/loader.js:778:12) {

code: 'MODULE_NOT_FOUND',

requireStack: [

'/home/suresh-dhwani/citifinance-node-api/node_modules/@azure/logger/dist/commonjs/log.js',

'/home/suresh-dhwani/citifinance-node-api/node_modules/@azure/logger/dist/commonjs/debug.js',

'/home/suresh-dhwani/citifinance-node-api/node_modules/@azure/logger/dist/commonjs/index.js',

'/home/suresh-dhwani/citifinance-node-api/node_modules/@azure/core-http/dist/index.js',

'/home/suresh-dhwani/citifinance-node-api/node_modules/@azure/storage-blob/dist/index.js',

'/home/suresh-dhwani/citifinance-node-api/service/blob_service.js',

'/home/suresh-dhwani/citifinance-node-api/service/getBlobUrl.js',

'/home/suresh-dhwani/citifinance-node-api/service/common.js',

'/home/suresh-dhwani/citifinance-node-api/service/index.js',

'/home/suresh-dhwani/citifinance-node-api/app.js',

'/home/suresh-dhwani/citifinance-node-api/server.js'

]

}

**Comment by RedYetiDev** [comment]
The comments for this issue seem to have begun to be unrelated

If you are experiencing an issue, please open an new issue, rather than commenting here

**Comment by estebancastano** [comment]
Thanks so much. This blog is very useful for me. The issue is resolved

**Comment by umejiburu** [comment]
how i solve this error

PS C:\Users\Dell\Desktop\projectss\dat> node index.js
Before requiring json-server
node:internal/modules/cjs/loader:1148
  throw err;
  ^

Error: Cannot find module 'json-server'
Require stack:
- C:\Users\Dell\Desktop\projectss\dat\index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Module._load (node:internal/modules/cjs/loader:986:27)       
    at Module.require (node:internal/modules/cjs/loader:1233:19)    
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (C:\Users\Dell\Desktop\projectss\dat\index.js:2:20)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)   
    at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)       
    at Module._load (node:internal/modules/cjs/loader:1024:12)      
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ 'C:\\Users\\Dell\\Desktop\\projectss\\dat\\index.js' ]
}

Node.js v20.15.0

**Comment by Trott** [comment]
> The comments for this issue seem to have begun to be unrelated
> 
> If you are experiencing an issue, please open an new issue, rather than commenting here

I'm locking this issue. Please open a new issue if you are experiencing a problem.
